perf-tool/src/agentOptions.cpp

```diff
@@ -105,18 +109,56 @@ void modifyMonitorEvents(const std::string& function, const std::string& command
             error = jvmti->AddCapabilities(&capa);
             check_jvmti_error(jvmti, error, "Unable to init Monitor Events capability.");
 
+            error = jvmti->SetEventNotificationMode(JVMTI_ENABLE, JVMTI_EVENT_MONITOR_CONTENDED_ENTER, (jthread)NULL);
+            check_jvmti_error(jvmti, error, "Unable to enable MonitorContendedEnter event notifications.");
             error = jvmti->SetEventNotificationMode(JVMTI_ENABLE, JVMTI_EVENT_MONITOR_CONTENDED_ENTERED, (jthread)NULL);
             check_jvmti_error(jvmti, error, "Unable to enable MonitorContendedEntered event notifications.");
         }
         else if (!command.compare("stop"))
         { 
             printf("Monitor Events Capability already disabled.");
+            error = jvmti->SetEventNotificationMode(JVMTI_DISABLE, JVMTI_EVENT_MONITOR_CONTENDED_ENTER, (jthread)NULL);
+            check_jvmti_error(jvmti, error, "Unable to disable MonitorContendedEnter event.");
             error = jvmti->SetEventNotificationMode(JVMTI_DISABLE, JVMTI_EVENT_MONITOR_CONTENDED_ENTERED, (jthread)NULL);
             check_jvmti_error(jvmti, error, "Unable to disable MonitorContendedEntered event.");
         } else {
             invalidCommand(function, command);
         }
     }
+    if (capa.can_get_monitor_info)
+    {
+        if (!command.compare("stop"))
+        {
+            memset(&capa, 0, sizeof(jvmtiCapabilities));
+            capa.can_get_monitor_info = 1;
+
+            error = jvmti->RelinquishCapabilities(&capa);
+            check_jvmti_error(jvmti, error, "Unable to relinquish Monitor Info Capability.");
+        }
+        else if (!command.compare("start"))
+        {
+            printf("Monitor Info Capability already enabled\n");
+        } else {
+            invalidCommand(function, command);
+        }
+    }
+    else
+    { /* cannot get monitor info */  
+        if (!command.compare("start"))
+        {
+            memset(&capa, 0, sizeof(jvmtiCapabilities));
+            capa.can_get_monitor_info = 1;
+
+            error = jvmti->AddCapabilities(&capa);
+            check_jvmti_error(jvmti, error, "Unable to init monitor info capability.");
```

Comment:
Maaybe it's better to group all this new code with the one before that deal with `con_generate_monitor_events` capability. I.e. set the two capabilities at the same time.